File: bblaser-service/src/main/java/nl/sijpesteijn/bblaserservice/services/AnimationService.java
package nl.sijpesteijn.bblaserservice.services;

import nl.sijpesteijn.bblaserservice.domain.BBAnimation;
import nl.sijpesteijn.bblaserservice.domain.BBAppearance;
import nl.sijpesteijn.bblaserservice.repositores.AnimationRepository;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import org.springframework.stereotype.Service;

import java.util.Optional;

@Service
public class AnimationService {

    private AnimationRepository animationRepository;

    public AnimationService(final AnimationRepository animationRepository) {
        this.animationRepository = animationRepository;
    }

    public Page<BBAnimation> getAnimations(final Pageable pageable) {
        return animationRepository.findAll(pageable);
    }

    public void save(BBAnimation animation) {
        animationRepository.save(animation);
    }

    public BBAnimation getAnimation(String id) {
        Optional<BBAnimation> bbAnimation = animationRepository.findById(id);
        if (bbAnimation.isPresent()) {
            return bbAnimation.get();
        }
        throw new IllegalArgumentException("Could not load animation with id " + id);
    }
}

File: bblaser-service/src/main/kotlin/nl/sijpesteijn/bblaser/services/AnimationService.kt
package nl.sijpesteijn.bblaser.services

import nl.sijpesteijn.bblaser.BBAnimation
import nl.sijpesteijn.bblaser.repositories.AnimationRepository
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.stereotype.Component
import java.util.*

@Component
class AnimationService(private val animationRepository: AnimationRepository) {
    fun all(pageable: Pageable): Page<BBAnimation> {
        return animationRepository.retrieveAllAnimationsPaged(pageable)
    }

    fun getById(id: String):Optional<BBAnimation> {
        return animationRepository.findById(id)
    }

    fun save(a: BBAnimation): BBAnimation {
        return animationRepository.save(a)
    }

    fun update(a: BBAnimation): BBAnimation {
        return animationRepository.save(a)
    }

    fun remove(id: String): Unit = animationRepository.deleteById(id)

}